frontend/src/utils/render_utils.jsx

```diff
@@ -0,0 +1,51 @@
+import { useLoader } from '@react-three/fiber';
+// import React from 'react';
+import * as THREE from 'three';
+import { GLTFLoader } from 'three/examples/jsm/loaders/GLTFLoader';
+
+const tileConfig = {
+    width: 1,
+    height: 0.25,
+};
+
+// Converts map coordinates to render coordinates.
+function convertToRenderCoordinates(mapVector, offset = {x: 0, y: 0, z: 0}) {
+    return new THREE.Vector3(
+        mapVector.x * tileConfig.width - offset.x,
+        mapVector.z * tileConfig.height,
+        mapVector.y * tileConfig.width - offset.y
+    )
+};
+
+// Produces the geometry for a map tile.
+export function renderTile(tile) {
+    const position = convertToRenderCoordinates(tile)
+    const spacer = 0.00; // add space between tiles to see better, remove after improved textures
+
+    return (
+                                                // place tiles below other objects
+        <mesh position={ [position.x, position.y - tileConfig.height/2, position.z] } key={`${tile.x},${tile.y},${tile.z}`}>
+            <boxGeometry args={ [tileConfig.width*(1-spacer), tileConfig.height*(1-spacer), tileConfig.width*(1-spacer)] } />
+            <meshBasicMaterial
+                color={ tile.type.material.color }
+            />
+        </mesh>
+    )
+};
+
+// Converts an object to a react-three-fiber component.
+export const objectToComponent = (obj) => {
```

Comment:
This one too. This should just be another component that the *View components would include. I put up a draft PR to show you where I was trying to go with those.

Reply:
https://github.com/project-brooklyn/project-brooklyn/pull/14

Reply:
Does everything (castle, portal, every type of tower) need its own "View Component"? If they are all converted into components in the same way, this makes more sense to me, but I still have to checkout your `<BaseView />`

Reply:
There doesn't necessarily have to be multiple components but if there will be things that are different between them, then it would make sense to add more on top. For example, the portal wouldn't have an hp bar (or maybe it will, hmmm).

I dunno if what I have currently will be sufficient to convince you yet. It might also be good to read up on the model -view-controller pattern.